.github/workflows/main-docs-deploy.yml

```diff
@@ -0,0 +1,46 @@
+name: "Documentation deployment"
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Pages
+        uses: actions/configure-pages@v5
+      - uses: ammaraskar/sphinx-action@dev
+        with:
+          docs-folder: "docs/"
+          pre-build-command: "pip install sphinx_autodoc_typehints"
```

Comment:
Can't we just use `pip install -r ...` in here? In case we add future dependencies to that, we won't need to change the workflow

Reply:
Fixed!